src/EnergyPlus/MixedAir.cc

```diff
@@ -569,7 +569,31 @@ void SimOAComponent(EnergyPlusData &state,
             OACoolingCoil = true;
         } else if (SELECT_CASE_var == DXSystem) { // CoilSystem:Cooling:DX  old 'AirLoopHVAC:UnitaryCoolOnly'
             if (Sim) {
-                SimDXCoolingSystem(state, CompName, FirstHVACIteration, AirLoopNum, CompIndex);
+                // SimDXCoolingSystem(state, CompName, FirstHVACIteration, AirLoopNum, CompIndex);
+                if (state.dataAirLoop->OutsideAirSys(OASysNum).compPointer[CompIndex] == nullptr) {
+                    UnitarySystems::UnitarySys thisSys;
+                    state.dataAirLoop->OutsideAirSys(OASysNum).compPointer[CompIndex] =
+                        thisSys.factory(state, DataHVACGlobals::UnitarySys_AnyCoilType, CompName, false, 0);
+                    UnitarySystems::UnitarySys::checkUnitarySysCoilInOASysExists(state, CompName, 0);
+                }
+                bool HeatingActive = false;
+                bool CoolingActive = false;
+                Real64 OAUCoilOutTemp = 0.0;
+                bool ZoneEquipFlag = false;
+                Real64 sensOut = 0.0;
+                Real64 latOut = 0.0;
+                state.dataAirLoop->OutsideAirSys(OASysNum).compPointer[CompIndex]->simulate(state,
+                                                                                            CompName,
+                                                                                            FirstHVACIteration,
+                                                                                            AirLoopNum,
+                                                                                            CompIndex,
+                                                                                            HeatingActive,
+                                                                                            CoolingActive,
+                                                                                            CompIndex,
+                                                                                            OAUCoilOutTemp,
+                                                                                            ZoneEquipFlag,
+                                                                                            sensOut,
+                                                                                            latOut);
```

Comment:
Typical factory and simulate call for OA System, PrimaryAirSystem, Zone equipment and OA Unit.